app/src/main/java/nu/yona/app/utils/AppUtils.java

```diff
@@ -288,25 +289,37 @@ public static String getHourDigit(String digit)
 	 */
 	public static void reportException(String className, Exception e, Thread t, DataLoadListener listener)
 	{
-		ErrorMessage errorMessage = new ErrorMessage();
-		if (e != null && e.getMessage() != null)
+		ErrorMessage errorMessage = getErrorMessage(e);
+		if (listener != null)
 		{
-			errorMessage.setMessage(e.getMessage());
+			listener.onError(errorMessage);
 		}
 		else
 		{
-			errorMessage.setMessage(YonaApplication.getAppContext().getString(R.string.error_message));
+			showErrorToast(errorMessage);
+			logExceptionToCrashlytics(e);
 		}
-		if (listener != null)
+	}
+
+	/**
+	 * getErrorMessage from give Exception or return a generic error message
+	 */
+
+	public static ErrorMessage getErrorMessage(Exception e)
```

Comment:
Generally don't use such short names. Exceptions to that rule are: lambda expressions, loop iterator value,  name of exception in catch block